packages/odf/components/utils/common.ts

```diff
@@ -370,27 +382,113 @@ export const getReplicasFromSelectedNodes = (
   }
   // When there are not enough zones/racks, we set the minimum replicas.
   // When there are more than 3 zones/racks, we set one OSD for each zone/rack.
-  return replicas < OCS_DEVICE_SET_MINIMUM_REPLICAS
+  return replicas < OCS_DEVICE_SET_MINIMUM_REPLICAS &&
+    !twoNodesOneArbiterCluster
     ? OCS_DEVICE_SET_MINIMUM_REPLICAS
     : replicas;
 };
```

Comment:
@aruniiird you need to also handle the case when TNA (OCS_TNA_DEVICE_SET_MINIMUM_REPLICAS to be created):
```suggestion
  const minimumReplicas = twoNodesOneArbiterCluster
    ? OCS_DEVICE_SET_MINIMUM_REPLICAS
    : OCS_TNA_DEVICE_SET_MINIMUM_REPLICAS;
  return replicas < minimumReplicas ? minimumReplicas : replicas;
```